doc/services/mem_mgmt/index.rst

```diff
@@ -84,7 +84,112 @@ one by renaming the property and changing its value according to the following l
    "IO"          -> <( DT_ARM_MPU(ATTR_MPU_IO) )>
    "EXTMEM"      -> <( DT_ARM_MPU(ATTR_MPU_EXTMEM) )>
 
+Memory Attributes Heap Allocator
+********************************
+
+It is possible to leverage the memory attribute property ``zephyr,memory-attr``
+to define and create a set of memory heaps from which the user can allocate
+memory from with certain attributes / capabilities.
+
+When the :kconfig:option:`CONFIG_MEM_ATTR_HEAP` is set, every region marked
+with one of the memory attributes listed in in
+:zephyr_file:`include/zephyr/dt-bindings/memory-attr/memory-attr-sw.h` is added
+to a pool of memory heaps used for dynamic allocation of memory buffers with
+certain attributes.
+
+Here a non exhaustive list of possible attributes:
+
+.. code-block:: none
+
+   DT_MEM_SW_ALLOC_CACHE
+   DT_MEM_SW_ALLOC_NON_CACHE
+   DT_MEM_SW_ALLOC_DMA
+
+For example we can define several memory regions with different attributes and
+use the appropriate attribute to indicate that it is possible to dynamically
+allocate memory from those regions:
+
+.. code-block:: devicetree
+
+   mem_cacheable: memory@10000000 {
+       compatible = "mmio-sram";
+       reg = <0x10000000 0x1000>;
+       zephyr,memory-attr = <( DT_MEM_CACHEABLE | DT_MEM_SW_ALLOC_CACHE )>;
+   };
+
+   mem_non_cacheable: memory@20000000 {
+       compatible = "mmio-sram";
+       reg = <0x20000000 0x1000>;
+       zephyr,memory-attr = <( DT_MEM_NON_CACHEABLE | ATTR_SW_ALLOC_NON_CACHE )>;
+   };
```

Comment:
ATTR_SW_ALLOC_NON_CACHE should be DT_MEM_SW_ALLOC_NON_CACHE?

Reply:
huuu, yeah, I'll fix that in a later commit. It would suck to lose all the ACKs to fix spelling errors.